mesonbuild/build.py

```diff
@@ -758,6 +758,8 @@ def __init__(
         self.pch: T.Dict[str, T.List[str]] = {}
         self.extra_args: T.DefaultDict[str, T.List[str]] = kwargs.get('language_args', defaultdict(list))
         self.sources: T.List[File] = []
+        # If the same source is defined multiple times, use it only once.
+        self.added_sources: T.Set[File] = set()
```

Comment:
Can we take this opportunity to rename it to "seen_sources"?

Reply:
I have no strong opinion on that, so sure, why not.